### PR TITLE
ci(lint-pr): reduce permissions to minimum

### PR DESCRIPTION
**Pull request by Fdawgs**
As discussed in https://github.com/nodejs/undici/pull/1356#discussion_r856063520, the use of `pull_request_target` here opens up a potential security hole.

Declaring the minimum [permissions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions) for this workflow to run, following principle of least privilege, negates this. See [related GitHub security post](https://securitylab.github.com/research/github-actions-building-blocks/) regarding permissions.

### CODE REVIEW

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by mcollina — APPROVED** [review]
lgtm

File: .github/workflows/lint-pr.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: 'Lint PR'
+name: 'Lint PR Title'
```
Comment (mcollina) [inline .github/workflows/lint-pr.yml:1]: Why this change?

**Review by mcollina — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-pr.yml (hunk 1 above)
Comment (Fdawgs) [reply .github/workflows/lint-pr.yml:1]: Workflow is only linting the PR title, not the entire PR (including contents), so updated it for clarity.

**Review by mcollina — APPROVED** [review]
lgtm
